doc/source/rllib/rllib-connector.rst

```diff
@@ -0,0 +1,186 @@
+.. include:: /_includes/rllib/announcement.rst
+
+.. include:: /_includes/rllib/we_are_hiring.rst
+
+Connectors
+==========
+
+Connector is a framework (currently in alpha status), introduced to solve a few common pain points related to the durabilty and
+deployment of RLlib's policy checkpoints.
+
+At the core of many RLlib algorithms are user environments and a decision making policy (usually a neural network).
+They interact with each other to produce training samples that can be used to improve the policy.
+
+Data observed from the environments usually go through multiple steps of preprocessing before they reach
+the policy, while the output of the policy also gets transformed multiple times before they are used to advance
+specific agents in the environments.
+
+By making sure these transformed are consolidated under the framework of connectors, users of RLlib will be able to:
+
+- Restore and deploy individual RLlib policies without having to restore training related logics of our Algorithms.
+- Make sure policies are more durable than the algorithms they get trained with.
+- Allow policies to be easily adapted to work with different versions of a user environment.
+- Run inference with RLlib polcies without worrying about the exact trajectory view requriements, or in the case of a recurrent policy, the state inputs.
+
+Connectors can be enabled by setting ``enable_connectors`` parameter to ``True``.
+
+Agent, Action, Lambda, and Pipeline Connectors
+------------------------------------------------------
+
+.. image:: images/connector-diagram.svg
+    :align: center
+
+Connectors transform data at two places. We name the ones transforming observation and action data
+``agent connectors`` and ``action connectors`` respectively.
+
+``Agent connectors`` handle the job of transforming environment observation data into a format that is understood by
+the policy (e.g., flattening arbitrary observation space into a flat tensor). The high level APIs are:
+
+.. code-block:: python
+
+    class AgentConnector(Connector):
+        def __call__(
+            self, acd_list: List[AgentConnectorDataType]
+        ) -> List[AgentConnectorDataType]:
+            ...
+
+        def reset(self, env_id: str):
+            ...
+
+        def on_policy_output(self, output: ActionConnectorDataType):
+            ...
+
+One thing to note about AgentConnector's API is that it works on a list of observation data.
+The list is constructed by grouping together observations from agents that are mapped to a same policy.
+
+This setup is useful for certain multi-agent use cases where individual observations may need to be
+modified based on data from other agents.
+This can also be useful if users need to construct meta-observation, e.g., build a graph as input
+to the policy from individual agent observations.
```

Comment:
Probably here you should add a paragraph explaining the `ActionConnectorDataType` data type.

Reply:
yeah, good idea, let me add a section about the common data types.